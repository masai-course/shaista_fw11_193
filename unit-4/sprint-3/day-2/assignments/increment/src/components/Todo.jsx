import { useState } from 'react';
function Todo(){
    const [text, setText]= useState("");
    const [todos, SetTodos]=useState([]);

    const handleChange =(e) =>{
        setText(e.target.value);
    }
    const handleAdd =()=>{
        SetTodos([...todos,text]);
        setText("")
    }
    
    return (
    <div>
    {todos.map((e)=>(
            <div> {e}</div>
        ))}
    <input
        value={text}
        onChange={handleChange}
        type="text"
        placeholder="Enter todo"
    
    
    />
    <button onClick={handleAdd}>Add todo</button>
    
    
    </div>
    )
}
export {Todo};