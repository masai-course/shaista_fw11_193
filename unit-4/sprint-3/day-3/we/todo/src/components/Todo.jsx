import { TodoTnput } from "./TodoInput";
import { useState } from "react";
import { TodoList } from "./TodoList";

export const Todos =()=>{
    const [todoList, setTodoList]= useState([]);

    const handleAddTodo = (todo) =>{
        setTodoList([...todoList,todo])
    }

 const handleToggle =(id)=>{
    const updatedList= todoList.map((e)=>{
if(todo.id==id){
    todo.states = !todoList
}
return todo;
     })

     setTodoList(updatedList);

 }
    return(
        <div>
            <TodoTnput handleAddTodo={handleAddTodo}/>
            <TodoList list={todoList} handleToggle={handleToggle}/>
        </div>
    )
}