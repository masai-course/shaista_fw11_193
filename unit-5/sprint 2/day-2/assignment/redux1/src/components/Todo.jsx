import {useDispatch,useSelector} from "react-redux";
import {useState,useEffect} from "react";
import {addTodo,addTodoLoading,addTodoSucess,addTodoError,getTodoLoading,getTodoSuccess,getTodoError} from "../redux/action";
import axios from "axios";


export const Todo = () => {
    useEffect(() =>{
        getData();
    },[])
    const getData = async() => {
        dispatch(getTodoLoading())
        try{
            const {data} = await axios.get("http://localhost:3001/todos")
            dispatch(getTodoSuccess(data));
        }catch(e){
            dispatch(getTodoError(e))
        }   
    }
    const dispatch=useDispatch();
    const {loading,error,data} = useSelector((store)=>store.todos)
    const [text,setText] = useState("");
    const [showtask,setShowtask] = useState(true);
    const handleChange = () =>{
        setText(text)
    }
    const handleToggle = id => {
        const updateTodo = data.map(item => item.id === id?{...item,status:!item.status}:item);
        dispatch(getTodoSuccess(updateTodo));
      } 
    return loading?("loading ..."):<div >
        <input type="text" placeholder="Enter the task" onChange={e=>setText(e.target.value)}/>
        <button  onClick={async()=>{
            dispatch(addTodoLoading())
            const payload={
                status:false,
                title:text
            };

            try{
                const {data} = await axios.post("http://localhost:3002/todos", payload)
                dispatch(addTodoSucess(data))

            }catch(e){
                dispatch(addTodoError(e));
            }
            //dispatch(addTodo(payload))
        }}>ADD</button>
        {
            data.filter(e=>showtask?true:!e.status).map(e=><div>
                {console.log(e.status)}
                <div>{e.title}</div>
                <button  onClick={()=>handleToggle(e.id)}>
                    <div style={{display:"flex",flexDirection:"row"}}><span style={{paddingTop:"20px"}}>{e.name}</span><span>{e.status? "Done":'Not Done'}</span></div>
                </button>
            </div>)
        }
        <br/>
        <button  onClick={()=>setShowtask(!showtask)}>{showtask?"SHOW NOT COMPELETED TASKS":"SHOW ALL TASK"}</button>
    </div>
}