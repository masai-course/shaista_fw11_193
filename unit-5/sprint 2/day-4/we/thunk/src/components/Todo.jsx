import {useDispatch,useSelector} from "react-redux";
import {useState,useEffect} from "react";
import {addTodoLoading,addTodoSucess,addTodoError,getTodoLoading,getTodoSuccess,getTodoError, getTodos} from "../redux/action";
import axios from "axios";

const styles = {
    width:"60%",
    height:"600px",
    border: "1px solid coral",
    backgroundColor:'skyblue',
    margin:"auto",
    marginTop:"10%"
  }

const textStyle = {
    fontSize:'30px',
    backgroundColor:'coral',
    width:"80%",
    height:'50px',
    border: "1px solid black",
    margin:"20px 10%",
    justifyContent:"space-between",
  }
  const inputStyle = {
    width:"70%",
    height:"50px",
    border:"none",
    fontSize:"30px",
    marginTop:"30px"
  }
  
  const btnStyle = {
    width:"100px",
    height:"50px",
    border:"none",
    marginLeft:"10px",
    paddingTop:"13px",
    paddingBottom:"11px",
    fontSize:"20px"
  }
export const Todo = () => {
    useEffect(() =>{
        getData();
    },[])
    const getData = async() => {
       dispatch(getTodos());
    }
    const dispatch=useDispatch();
    const {loading,error,data} = useSelector((store)=>store.todos)
    const [text,setText] = useState("");
    const [showtask,setShowtask] = useState(true);
    const handleChange = () =>{
        setText(text)
    }
    const handleToggle = id => {
        const updateTodo = data.map(item => item.id === id?{...item,status:!item.status}:item);
        dispatch(getTodoSuccess(updateTodo));
      } 
    return loading?("loading ..."):<div sytle={styles}>
        <input style={inputStyle} type="text" placeholder="Enter the task" onChange={e=>setText(e.target.value)}/>
        <button style={btnStyle} onClick={async()=>{
            dispatch(addTodoLoading())
            const payload={
                status:false,
                title:text
            };

            try{
                const {data} = await axios.post("http://localhost:3001/todos", payload)
                dispatch(addTodoSucess(data))

            }catch(e){
                dispatch(addTodoError(e));
            }
            //dispatch(addTodo(payload))
        }}>ADD</button>
        {
            data.filter(e=>showtask?true:!e.status).map(e=><div>
                {console.log(e.status)}
                <div>{e.title}</div>
                <button style={textStyle} onClick={()=>handleToggle(e.id)}>
                    <div style={{display:"flex",flexDirection:"row"}}><span style={{paddingTop:"20px"}}>{e.name}</span><span style={{paddingLeft:"330px",textAlign:"right",float:"right"}}>{e.status? "Done":'Not Done'}</span></div>
                </button>
            </div>)
        }
        <button style={{width: '80%',margin:'auto',height:"50px",fontSize:"30px",backgroundColor:"#00838F"}} onClick={()=>setShowtask(!showtask)}>{showtask?"SHOW NOT COMPELETED TASKS":"SHOW ALL TASK"}</button>
    </div>
}