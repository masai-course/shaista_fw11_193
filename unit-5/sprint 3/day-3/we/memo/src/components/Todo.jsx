import { useState } from "react"

export const Todos =() =>{
    const [text,setText] = useState("");
    const [todoList,setTodolist] = useState([]);
    const handleToggle =  useCallback ((id)=>{
        cosole.log(id);
    },[])
return (
    <div>
    <input
    type="text"
    value = 
    </div>
)
    
}