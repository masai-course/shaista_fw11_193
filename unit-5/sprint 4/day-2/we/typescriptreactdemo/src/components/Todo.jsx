import { useState } from "react"
import { TodoInput } from "./TodoInput";

export const Todo = () =>{

    const [text,setText] = useState("");
    const [todos,setTodos] = useState([]);
    return <div>
    
    <input type="text" placeholder="Enter todo" onChange= {e => setText(e.target.value)} />
    <button onClick={()=>{
        setTodos([...todos,text])
    }}>add todo</button>
    {todos.map((e,i) => <TodoInput key={i} title={e}/>)}
    </div>
}