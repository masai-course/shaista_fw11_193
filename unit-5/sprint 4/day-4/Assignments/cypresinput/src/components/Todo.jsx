import {useState} from "react";
export const Todo = () => {
    const [text,setText] = useState("");
    const [data,setData] = useState([]);
    const handleAdd = () => {
        const temp = [...data,text]
        setData(temp)
    }
    data.className="todoitems"
    console.log(data);
    return (<div>
        <input className="todoinput" onChange={(e)=>{setText(e.target.value)}}placeholder="Enter the task name"/>
        <button className="todobtn" onClick={handleAdd}>Add</button>
        {
            data.map(e=><div>{e}</div>)
        }
    </div>)
}