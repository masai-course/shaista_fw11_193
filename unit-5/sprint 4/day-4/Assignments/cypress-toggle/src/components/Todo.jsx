import {useState} from "react";
export const Todo = () => {
    const [text,setText] = useState("");
    const [data,setData] = useState([]);
    const handleAdd = () => {
        setData([...data,{title:text,status:"false"}])
    }
    const handlecheckbox=(value) => {
        let elem = document.getElementById("textbox");
        let docelem = document.getElementById("text");
        elem.textContent = value.target.checked;
        value.target.checked?docelem.style.backgroundColor="green":docelem.style.backgroundColor="white";
    }
    return (<div>
        <h1>Welcome to the Todo Project</h1>
        <input className="inp" type="text" placeholder="Enter the Task Name" onChange={(e) => setText(e.target.value)}/>
        <button className="btn" onClick={handleAdd}>Add</button>
        {
            data.map((e,i)=>{
                return <div key={i} style={{display:"flex", flexDirection:"row",marginLeft:"350px"}}>
                    <div id="text">{e.title}</div>
                    <input type="checkbox" value={e.status} onClick={handlecheckbox}/>
                    <div id="textbox"></div>
                </div>
            })
        }
    </div>)
}