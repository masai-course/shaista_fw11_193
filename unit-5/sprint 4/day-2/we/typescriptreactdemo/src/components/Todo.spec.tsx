import {render, screen, fireEvent} from "@testing-library/react";
import {Todo} from "./Todo";
test("todos in document", function(){
    const {container} = render(<Todo/>)
    expect(container).toMatchSnapshot()
})