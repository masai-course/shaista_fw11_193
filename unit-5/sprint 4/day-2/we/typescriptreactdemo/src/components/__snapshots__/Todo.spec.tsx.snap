// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todos in document 1`] = `
<div>
  <div>
    <input
      placeholder="Enter todo"
      type="text"
    />
    <button>
      add todo
    </button>
  </div>
</div>
`;
